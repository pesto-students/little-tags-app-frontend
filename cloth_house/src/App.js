import "./App.css";
import Routes from "./route";
import ErrorBoundary from "./Hoc/ErrorBoundary";
import { BrowserRouter } from "react-router-dom";
import {IntlProvider} from 'react-intl';
import messages from "./messages";
function App(props) {
 const locale =  'fr-FR'; 
 
  return (

    <div className="App">
    
      <BrowserRouter>
        <ErrorBoundary>
     <IntlProvider locale={locale} messages={messages[locale]}>
          <Routes {...props} />
          </IntlProvider>
        </ErrorBoundary>
      </BrowserRouter>
    </div>
  );
}

export default App;
