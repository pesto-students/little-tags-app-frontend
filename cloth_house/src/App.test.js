import React from 'react';
import App from './App';
import renderer from 'react-test-renderer';
// import ErrorBoundary from "./Hoc/ErrorBoundary";
// import Routes from "./route";
// import { IntlProvider, intlShape } from "react-intl";

// Create IntlProvider to retrieve React Intl context
// const intlProvider = new IntlProvider(
//   {
//     locale: "en",
//     messages: {
//       message1: "Hello world"
//     }
//   },
//   {}
// );
// const { intl } = intlProvider.getChildContext();

// // `intl` prop is required when using injectIntl HOC
// const nodeWithIntlProp = node => React.cloneElement(node, { intl });

// // shallow() with React Intl context
// global.shallowWithIntl = (node, { context, ...options } = {}) => {
//   return shallow(nodeWithIntlProp(node), {
//     ...options,
//     context: {
//       ...context,
//       intl
//     }
//   });
// };
// // mount() with React Intl context
// global.mountWithIntl = (
//   node,
//   { context, childContextTypes, ...options } = {}
// ) => {
//   return mount(nodeWithIntlProp(node), {
//     ...options,
//     context: {
//       ...context,
//       intl
//     },
//     childContextTypes: {
//       intl: intlShape,
//       ...childContextTypes
//     }
//   });
// };

it('Route Snapshot', () => {
  const tree = renderer.create(<App />).toJSON();
  expect(tree).toMatchSnapshot();
});


//module.exports = Intl;


// it('Error Boundary Snapshot', () => {
//   const tree = renderer.create(<ErrorBoundary />).toJSON();
//   expect(tree).toMatchSnapshot();
// });