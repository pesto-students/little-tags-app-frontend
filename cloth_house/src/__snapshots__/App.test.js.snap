// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Route Snapshot 1`] = `
<div
  className="App"
>
  <div
    style={
      Object {
        "alignItems": "center",
        "color": "#FF7574",
        "display": "flex",
        "justifyContent": "center",
        "margin": "0px 0px 10px 10px",
        "width": "100%",
      }
    }
  >
    <h2
      className="centered"
      style={
        Object {
          "fontSize": "16px",
        }
      }
    >
      Whoops! Something went wrong. We will be looking into this.
    </h2>
  </div>
</div>
`;
